src/output.rs

```diff
@@ -440,3 +440,51 @@ impl PartialEq<Output> for WeakOutput {
         self.upgrade().map(|o| &o == other).unwrap_or(false)
     }
 }
+
+/// Source for determining output mode information.
+#[derive(PartialEq, Clone, Debug)]
+pub enum OutputModeSource {
```

Comment:
Moved all this to `output`, I think that's a much better place for it anyway (and the file is smaller too).